src/lib/optimizer/optimizer.cpp

```diff
@@ -21,10 +21,10 @@ Optimizer Optimizer::create_default_optimizer() {
   RuleBatch main_batch(RuleBatchExecutionPolicy::Iterative);
   main_batch.add_rule(std::make_shared<PredicateReorderingRule>());
   main_batch.add_rule(std::make_shared<JoinDetectionRule>());
-  main_batch.add_rule(std::make_shared<ChunkPruningRule>());
   optimizer.add_rule_batch(main_batch);
 
   RuleBatch final_batch(RuleBatchExecutionPolicy::Once);
+  final_batch.add_rule(std::make_shared<ChunkPruningRule>());
```

Comment:
I don't know how happy I am with that. The reason: a strongly pruning predicate could yield better plans; we should use that information during predicate reordering.

What do you guys think about that?

Reply:
As discussed today: let's not dig deeper into this issue.